### PR TITLE
Add OrderProcessingWorkflow and couple of enhancements for the workflows

### PR DESCRIPTION
**Pull request by gazi-yestemirova**
Adding OrderProcessingWorkflow as a second example for students to create activities. 
As well as couple of tasks to fail a workflow and use retry policies.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,5 +65,63 @@ To start the hello world workflow, run the following command:
...
+
+### Task 2: Simulate Payment Failure in validatePayment() activity
+First, modify the processPaymentActivity to simulate a failure. This can be done by causing the activity to fail intermittently, such as by returning an error during the first few attempts, which will simulate a temporary issue like a network failure.
+<br />
+<br />
+Use the code below to simulate the failure:
+```go
+func processPaymentActivity(ctx context.Context, order Order) (string, error) {
+	// Simulate a failure
+	if time.Now().Unix()%2 == 0 { // Fail intermittently (every second attempt).
```
Comment (dkrotx) [inline README.md]: maybe more explicitly - to take random?

File: README.md [hunk 2]
```diff
@@ -65,5 +65,63 @@ To start the hello world workflow, run the following command:
...
+	return "Payment successful", nil
+}
+```
+
+Ensure that the workflow fails intermittently and return an error.
+
+### Task 3: Add Retry Policy to validatePayment() activity
+Now, implement a retry policy to handle the temporary failures simulated in Task 2. <br /> 
+The retry policy should retry the validatePayment activity up to 3 times with exponential backoff. <br />
+Use the paymentRetryPolicy configuration below:
```
Comment (dkrotx) [inline README.md:106]: I didn't follow this completely, but isn't `processPaymentActivity` has 50% chance of just working? If so, half of students may not even face with the issue. Maybe to drop this probability substantially?

File: workflow/postnord/postnord.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
+package postnord
```
Comment (dkrotx) [inline workflow/postnord/postnord.go]: nit: maybe call the file after what is doing, not just `postnord.go`?

File: workflow/postnord/postnord.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+func RegisterWorkflow(w worker.Worker) {
+	w.RegisterWorkflowWithOptions(OrderProcessingWorkflow, workflow.RegisterOptions{Name: WorkflowName})
+	// Register your activities here
+}
+
+// Order represents an order with basic details like the ID, customer name, and order amount.
+type Order struct {
+	ID       string  `json:"id"`
+	Customer string  `json:"customer"`
+	Amount   float64 `json:"amount"`
```
Comment (dkrotx) [inline workflow/postnord/postnord.go]: amount of goods or money? Can it be fractional?

Comment (gazi-yestemirova) [reply workflow/postnord/postnord.go]: it is an order amount, it refers to the total sum of the prices for all items in an order

**Review by dkrotx — APPROVED** [review]

**Pull request merged**